lib/src/v2/helpers.ts

```diff
@@ -64,9 +63,6 @@ export function genId(info: Uint8Array,
   const hash1 = crypto_hash_sha256(
       Uint8Array.from([...info, ...nonce1]) );
 
-  return crypto_hash_sha256(
-      IBEIdentityInternal.encode(
-          IBEIdentityInternal.create({hash: hash1, counter, nonce: nonce2}),
-      ).finish(),
-  );
+  return crypto_hash_sha256(Uint8Array.from(
+      [...hash1, ...from_string(counter.toString()), ...nonce2]));
```

Comment:
change order so this is more robust: fixed width arguments first, so for example `hash1, nonce2, counter`